pkg/operator/ceph/csi/spec.go

```diff
@@ -428,3 +433,51 @@ func createCSIDriverInfo(clientset kubernetes.Interface, name string) error {
 
 	return err
 }
+
+// ValidateCSIVersion checks if the configured ceph-csi image is supported
+func ValidateCSIVersion(clientset kubernetes.Interface, namespace, csiPluginImage, rookImage, serviceAccountName string) error {
+	var ownerRef metav1.OwnerReference
+	timeout := 15 * time.Minute
+
+	logger.Infof("detecting the ceph csi image version for image %s", csiPluginImage)
+
+	pod, err := k8sutil.GetRunningPod(clientset)
+	if len(pod.GetOwnerReferences()) == 0 {
+		return errors.Errorf("Could not parse the owner of the rook operator")
+	}
+	ownerRef = pod.GetOwnerReferences()[0]
+
+	versionReporter, err := cmdreporter.New(
+		clientset,
+		&ownerRef,
+		detectCSIVersionName, detectCSIVersionName, namespace,
+		[]string{"cephcsi"}, []string{"--version"},
+		rookImage, csiPluginImage)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to set up ceph CSI version job.")
```

Comment:
the error message should not end with punctuation as per https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
this applies to other places also in this PR

Reply:
Thank you for the pointer! I will change that.